### PR TITLE
#40: Fix formatting issues so code passes new pylint check

### LINKED ISSUE #40: Fix formatting issues so code passes new pylint check
Related to #15 (PR #19)

### PR DESCRIPTION
**Pull request by cwschilly**
Fixes #40 
Related to PR #19

### CODE REVIEW

**Comment by cwschilly** [comment]
This PR is also related to the documentation PRs (33-39) since some pylint errors come from the length of the docstrings. I've made edits in the corresponding PRs to resolve those errors.

File: romtools/hyper_reduction/deim.py [hunk 1]
```diff
@@ -45,84 +45,79 @@
...
+    '''
+    Version of DEIM for multi-state systems.
+    We perform DEIM on each state variable, and
+    then return the union of all indices.
+    Repeated indices are removed.
+    '''
+    all_indices = np.zeros(0,dtype=int)
+    for i in range(0,n_var):
+        dataMatrix = getDataMatrixForIthVar(i,n_var,U,variable_ordering)
+        indices = deim(dataMatrix)
```
[the remark this answers is not in the document]
Comment (fnrizzi) [reply romtools/hyper_reduction/deim.py:91]: can you please rebase and try again? 
that function is unused for now, so i commented it out

[the remark this answers is not in the document]
Comment (cwschilly) [reply romtools/hyper_reduction/deim.py]: This worked, thanks!

**Comment by cwschilly** [comment]
@fnrizzi To see the pylint check passing, see the checks on the [penultimate commit](https://github.com/Pressio/rom-tools-and-workflows/actions/runs/6641117596/job/18042912825) (fe84aa5fb48c17f2ea7111a1619aeb31d7c0a778)

**Comment by fnrizzi** [comment]
i see a bunch of stuff like this:
```
Successfully installed astroid-3.0.1 dill-0.3.7 isort-5.12.0 mccabe-0.7.0 platformdirs-3.11.0 pylint-3.0.2 tomlkit-0.12.1 typing-extensions-4.8.0
************* Module romtools
romtools/__init__.py:113:0: C0301: Line too long (216/130) (line-too-long)
************* Module romtools.trial_space
romtools/trial_space.py:230:0: C0301: Line too long (163/130) (line-too-long)
romtools/trial_space.py:2[43](https://github.com/Pressio/rom-tools-and-workflows/actions/runs/6641117596/job/18042912825#step:9:44):8: C0103: Variable name "svdPicked" doesn't conform to snake_case naming style (invalid-name)
************* Module romtools.hyper_reduction.deim
romtools/hyper_reduction/deim.py:74:17: W1401: Anomalous backslash in string: '\P'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
romtools/hyper_reduction/deim.py:75:77: W1401: Anomalous backslash in string: '\P'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
romtools/hyper_reduction/deim.py:1:0: C0114: Missing module docstring (missing-module-docstring)
romtools/hyper_reduction/deim.py:48:0: W0105: String statement has no effect (pointless-string-statement)
romtools/hyper_reduction/deim.py:50:0: C0103: Function name "__getDataMatrixForIthVar" doesn't conform to camelCase naming style (invalid-name)
romtools/hyper_reduction/deim.py:52:4: R1705: Unnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return)
romtools/hyper_reduction/deim.py:50:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
romtools/hyper_reduction/deim.py:65:4: C0103: Variable name "sampledFunctionBasis" doesn't conform to snake_case naming style (invalid-name)
romtools/hyper_reduction/deim.py:66:4: C0103: Variable name "PU_pinv" doesn't conform to snake_case naming style (invalid-name)
romtools/hyper_reduction/deim.py:76:4: C0103: Variable name "sampledFunctionBasis" doesn't conform to snake_case naming style (invalid-name)
romtools/hyper_reduction/deim.py:77:4: C0103: Variable name "PU_pinv" doesn't conform to snake_case naming style (invalid-name)
************* Module romtools.trial_space_utils.scaler
romtools/trial_space_utils/scaler.py:120:4: R1705: Unnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return)
romtools/trial_space_utils/scaler.py:104:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
romtools/trial_space_utils/scaler.py:254:15: R1714: Consider merging these comparisons with 'in' by using 'variable_ordering in ('C', 'F')'. Use a set instead if elements are hashable. (consider-using-in)
romtools/trial_space_utils/scaler.py:267:8: W0201: Attribute 'var_scales_' defined outside __init__ (attribute-defined-outside-init)
romtools/trial_space_utils/scaler.py:331:8: C0103: Attribute name "__myVariableScaler" doesn't conform to snake_case naming style (invalid-name)
romtools/trial_space_utils/scaler.py:332:8: C0103: Attribute name "__myVectorScaler" doesn't conform to snake_case naming style (invalid-name)
************* Module romtools.trial_space_utils.splitter
romtools/trial_space_utils/splitter.py:135:15: R1714: Consider merging these comparisons with 'in' by using 'variable_ordering in ('C', 'F')'. Use a set instead if elements are hashable. (consider-using-in)
romtools/trial_space_utils/splitter.py:196:4: R1705: Unnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return)
romtools/trial_space_utils/splitter.py:180:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
************* Module romtools.trial_space_utils.orthogonalizer
```

is it expected?

**Comment by cwschilly** [comment]
@fnrizzi In short, yes. Some warnings are not resolved because I wasn't sure which of they were significant to the project. Rather than disable them or change the code to resolve them, I wanted to see what you or others had to say. 

For example, I fixed most variable names but some (the remaining cases) may be unconventional on purpose. I decided to let pylint throw the warning rather than cluttering the code with `# pylint: disable` statements. 

Other warnings I didn't fully understand: for example `nnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return)`. This is in reference to blocks like:

```python
def __getDataMatrixForIthVar(i,n_var,data_matrix,variable_ordering):
    '''helper function to split data'''
    if variable_ordering == 'F':
        return data_matrix[i::n_var]
    elif variable_ordering == 'C':
        n = int( data_matrix.shape[0] / n_var)
        start_index = i*n
        end_index = (i+1)*n
        return data_matrix[start_index:end_index]
```

In this case, I can easily switch from `elif` to `if`, but I don't see how that improves the code.

I can disable these warnings in `.pylintrc` or with `# pylint: disable-<warning>`, but I didn't want to accidentally disable something useful.

**Comment by fnrizzi** [comment]
can you please rebase this and then post here the full output of the lint stage so i can look at it? thanks

**Comment by cwschilly** [comment]
@fnrizzi The new pylint output is:

```
<unknown>:71: SyntaxWarning: invalid escape sequence '\P'
************* Module romtools
romtools/__init__.py:113:0: C0301: Line too long (216/130) (line-too-long)
************* Module romtools.trial_space
romtools/trial_space.py:230:0: C0301: Line too long (163/130) (line-too-long)
romtools/trial_space.py:243:8: C0103: Variable name "svdPicked" doesn't conform to snake_case naming style (invalid-name)
************* Module romtools.trial_space_utils.svd_method_of_snapshots
romtools/trial_space_utils/svd_method_of_snapshots.py:67:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
romtools/trial_space_utils/svd_method_of_snapshots.py:75:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
romtools/trial_space_utils/svd_method_of_snapshots.py:105:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
romtools/trial_space_utils/svd_method_of_snapshots.py:110:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
romtools/trial_space_utils/svd_method_of_snapshots.py:166:0: C0305: Trailing newlines (trailing-newlines)
romtools/trial_space_utils/svd_method_of_snapshots.py:1:0: C0114: Missing module docstring (missing-module-docstring)
romtools/trial_space_utils/svd_method_of_snapshots.py:49:11: I1101: Module 'mpi4py.MPI' has no 'COMM_WORLD' member, but source is unavailable. Consider adding this module to extension-pkg-allow-list if you want to perform analysis based on run-time introspection of living objects. (c-extension-no-member)
romtools/trial_space_utils/svd_method_of_snapshots.py:54:4: C0103: Constant name "mpi_rank" doesn't conform to UPPER_CASE naming style (invalid-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:55:4: C0103: Constant name "num_processes" doesn't conform to UPPER_CASE naming style (invalid-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:59:0: C0103: Function name "A_transpose_dot_bImpl" doesn't conform to camelCase naming style (invalid-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:59:32: W0621: Redefining name 'comm' from outer scope (line 49) (redefined-outer-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:64:4: W0621: Redefining name 'mpi_rank' from outer scope (line 50) (redefined-outer-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:65:4: W0621: Redefining name 'num_processes' from outer scope (line 51) (redefined-outer-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:74:4: C0103: Variable name "ATb_glob" doesn't conform to snake_case naming style (invalid-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:81:4: R1705: Unnecessary "else" after "return", remove the "else" and de-indent the code inside it (no-else-return)
romtools/trial_space_utils/svd_method_of_snapshots.py:87:40: W0621: Redefining name 'comm' from outer scope (line 49) (redefined-outer-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:94:4: C0103: Variable name "STS" doesn't conform to snake_case naming style (invalid-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:95:4: C0103: Variable name "Lam" doesn't conform to snake_case naming style (invalid-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:95:8: C0103: Variable name "E" doesn't conform to snake_case naming style (invalid-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:97:4: C0103: Variable name "U" doesn't conform to snake_case naming style (invalid-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:119:0: C0103: Class name "svdMethodOfSnapshots" doesn't conform to PascalCase naming style (invalid-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:147:22: W0621: Redefining name 'comm' from outer scope (line 49) (redefined-outer-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:151:8: C0103: Variable name "U" doesn't conform to snake_case naming style (invalid-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:155:0: C0103: Class name "svdMethodOfSnapshotsForQr" doesn't conform to PascalCase naming style (invalid-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:159:23: W0621: Redefining name 'comm' from outer scope (line 49) (redefined-outer-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:163:8: C0103: Variable name "U" doesn't conform to snake_case naming style (invalid-name)
romtools/trial_space_utils/svd_method_of_snapshots.py:163:11: W0612: Unused variable 's' (unused-variable)
************* Module romtools.trial_space_utils.outputter
romtools/trial_space_utils/outputter.py:1:0: C0114: Missing module docstring (missing-module-docstring)
romtools/trial_space_utils/outputter.py:62:0: C0103: Function name "npz_output" doesn't conform to camelCase naming style (invalid-name)
romtools/trial_space_utils/outputter.py:81:0: C0103: Function name "hdf5_output" doesn't conform to camelCase naming style (invalid-name)
romtools/trial_space_utils/outputter.py:96:0: C0103: Function name "exodus_ouput" doesn't conform to camelCase naming style (invalid-name)
************* Module romtools.trial_space_utils.scaler
romtools/trial_space_utils/scaler.py:120:4: R1705: Unnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return)
romtools/trial_space_utils/scaler.py:104:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
romtools/trial_space_utils/scaler.py:254:15: R1714: Consider merging these comparisons with 'in' by using 'variable_ordering in ('C', 'F')'. Use a set instead if elements are hashable. (consider-using-in)
romtools/trial_space_utils/scaler.py:267:8: W0201: Attribute 'var_scales_' defined outside __init__ (attribute-defined-outside-init)
romtools/trial_space_utils/scaler.py:331:8: C0103: Attribute name "__myVariableScaler" doesn't conform to snake_case naming style (invalid-name)
romtools/trial_space_utils/scaler.py:332:8: C0103: Attribute name "__myVectorScaler" doesn't conform to snake_case naming style (invalid-name)
************* Module romtools.trial_space_utils.splitter
romtools/trial_space_utils/splitter.py:135:15: R1714: Consider merging these comparisons with 'in' by using 'variable_ordering in ('C', 'F')'. Use a set instead if elements are hashable. (consider-using-in)
romtools/trial_space_utils/splitter.py:196:4: R1705: Unnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return)
romtools/trial_space_utils/splitter.py:180:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
************* Module romtools.trial_space_utils.orthogonalizer
romtools/trial_space_utils/orthogonalizer.py:95:0: C0301: Line too long (144/130) (line-too-long)
romtools/trial_space_utils/orthogonalizer.py:122:0: C0301: Line too long (144/130) (line-too-long)
romtools/trial_space_utils/orthogonalizer.py:101:8: C0103: Attribute name "__qrPicked" doesn't conform to snake_case naming style (invalid-name)
romtools/trial_space_utils/orthogonalizer.py:129:8: C0103: Attribute name "__qrPicked" doesn't conform to snake_case naming style (invalid-name)
************* Module romtools.hyper_reduction.deim
romtools/hyper_reduction/deim.py:74:17: W1401: Anomalous backslash in string: '\P'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
romtools/hyper_reduction/deim.py:75:77: W1401: Anomalous backslash in string: '\P'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
romtools/hyper_reduction/deim.py:1:0: C0114: Missing module docstring (missing-module-docstring)
romtools/hyper_reduction/deim.py:48:0: W0105: String statement has no effect (pointless-string-statement)
romtools/hyper_reduction/deim.py:50:0: C0103: Function name "__getDataMatrixForIthVar" doesn't conform to camelCase naming style (invalid-name)
romtools/hyper_reduction/deim.py:52:4: R1705: Unnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return)
romtools/hyper_reduction/deim.py:50:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
romtools/hyper_reduction/deim.py:65:4: C0103: Variable name "sampledFunctionBasis" doesn't conform to snake_case naming style (invalid-name)
romtools/hyper_reduction/deim.py:66:4: C0103: Variable name "PU_pinv" doesn't conform to snake_case naming style (invalid-name)
romtools/hyper_reduction/deim.py:76:4: C0103: Variable name "sampledFunctionBasis" doesn't conform to snake_case naming style (invalid-name)
romtools/hyper_reduction/deim.py:77:4: C0103: Variable name "PU_pinv" doesn't conform to snake_case naming style (invalid-name)
************* Module romtools.workflows.parameter_spaces
romtools/workflows/parameter_spaces.py:52:0: W0105: String statement has no effect (pointless-string-statement)
************* Module romtools.workflows.__init__
romtools/workflows/__init__.py:47:0: C0301: Line too long (132/130) (line-too-long)
************* Module romtools.workflows.dakota.sampling
romtools/workflows/dakota/sampling.py:64:4: W0238: Unused private member `DakotaSamplingCouplerBase.__base_directory` (unused-private-member)
romtools/workflows/dakota/sampling.py:108:0: C0103: Function name "run_dakota_sampling" doesn't conform to camelCase naming style (invalid-name)
************* Module romtools.workflows.greedy.run_greedy
romtools/workflows/greedy/run_greedy.py:88:0: R0914: Too many local variables (33/20) (too-many-locals)
romtools/workflows/greedy/run_greedy.py:100:4: C0103: Variable name "parameterSpace" doesn't conform to snake_case naming style (invalid-name)
romtools/workflows/greedy/run_greedy.py:92:18: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
romtools/workflows/greedy/run_greedy.py:88:0: R0915: Too many statements (127/50) (too-many-statements)
romtools/workflows/greedy/run_greedy.py:301:0: C0103: Class name "qoi_vs_error_indicator_regressor" doesn't conform to PascalCase naming style (invalid-name)
************* Module romtools.workflows.greedy.greedy_coupler_base
romtools/workflows/greedy/greedy_coupler_base.py:60:0: C0115: Missing class docstring (missing-class-docstring)
romtools/workflows/greedy/greedy_coupler_base.py:83:8: W0199: Assert called on a populated tuple. Did you mean 'assert x,y'? (assert-on-tuple)
romtools/workflows/greedy/greedy_coupler_base.py:176:4: C0103: Method name "__setupRomCase" doesn't conform to camelCase naming style (invalid-name)
romtools/workflows/greedy/greedy_coupler_base.py:182:4: C0103: Method name "__setupFomCase" doesn't conform to camelCase naming style (invalid-name)
************* Module romtools.workflows.greedy
romtools/workflows/greedy/__init__.py:1:0: C0114: Missing module docstring (missing-module-docstring)
************* Module romtools.workflows.sampling.sampling_coupler_base
romtools/workflows/sampling/sampling_coupler_base.py:134:4: C0103: Method name "__setupCase" doesn't conform to camelCase naming style (invalid-name)
************* Module romtools.workflows.sampling.sampling
romtools/workflows/sampling/sampling.py:1:0: C0114: Missing module docstring (missing-module-docstring)
romtools/workflows/sampling/sampling.py:57:4: C0103: Variable name "parameterSpace" doesn't conform to snake_case naming style (invalid-name)
romtools/workflows/sampling/sampling.py:59:4: W0612: Unused variable 'n_params' (unused-variable)
************* Module romtools.workflows.sampling.__init__
romtools/workflows/sampling/__init__.py:1:0: R0801: Similar lines in 2 files
==romtools.workflows.greedy.greedy_coupler_base:[124:134]
==romtools.workflows.sampling.sampling_coupler_base:[114:123]
        n_samples = parameter_samples.shape[0]

        path_to_work_dir = self.__base_directory + '/' + self.__work_directory_base_name + '/'
        if os.path.isdir(path_to_work_dir):
            pass
        else:
            os.mkdir(path_to_work_dir)

        for sample_no in range(starting_sample_no,starting_sample_no + n_samples):
            path_to_dir = ( (duplicate-code)
romtools/workflows/sampling/__init__.py:1:0: R0801: Similar lines in 2 files
==romtools.hyper_reduction.deim:[51:59]
==romtools.trial_space_utils.splitter:[195:202]
    if variable_ordering == 'F':
        return data_matrix[i::n_var]
    elif variable_ordering == 'C':
        n = int( data_matrix.shape[0] / n_var)
        start_index = i*n
        end_index = (i+1)*n
        return data_matrix[start_index:end_index]
 (duplicate-code)
romtools/workflows/sampling/__init__.py:1:0: R0801: Similar lines in 2 files
==romtools.trial_space_utils.orthogonalizer:[63:72]
==romtools.trial_space_utils.splitter:[89:98]
    @abc.abstractmethod
    def __init__(self) -> None:
        pass

    @abc.abstractmethod
    def __call__(self, my_array: np.ndarray) -> np.ndarray:
        pass

 (duplicate-code)

------------------------------------------------------------------
Your code has been rated at 9.03/10 (previous run: 0.43/10, +8.60)
```

File: romtools/trial_space_utils/svd_method_of_snapshots.py [hunk 2]
```diff
@@ -47,12 +47,11 @@
...
     comm = MPI.COMM_WORLD
-    mpi_rank = comm.Get_rank()
-    num_processes = comm.Get_size()
 except ModuleNotFoundError:
     print("module 'mpi4py' is not installed")
-    mpi_rank = 0
-    num_processes = 1
+
+mpi_rank = 0
+num_processes = 1
```
Comment (fnrizzi) [inline romtools/trial_space_utils/svd_method_of_snapshots.py:54]: why are we doing this?

Comment (cwschilly) [reply romtools/trial_space_utils/svd_method_of_snapshots.py:54]: It was throwing this error:
```
romtools/trial_space_utils/svd_method_of_snapshots.py:101:7: E0601: Using variable 'num_processes' before assignment (used-before-assignment)
romtools/trial_space_utils/svd_method_of_snapshots.py:106:7: E0601: Using variable 'mpi_rank' before assignment (used-before-assignment)
```

By bringing them into the global scope it resolved all errors/warnings. If I understand the code correctly, the functions will redefine these variables later if MPI is being used.

Comment (fnrizzi) [reply romtools/trial_space_utils/svd_method_of_snapshots.py:54]: i personally do not like global variables. can we change the code to avoid global?

Comment (cwschilly) [reply romtools/trial_space_utils/svd_method_of_snapshots.py:54]: Yes let me try other solutions

Comment (fnrizzi) [reply romtools/trial_space_utils/svd_method_of_snapshots.py]: yes it is an impl function anyway so lets do that

Comment (fnrizzi) [reply romtools/trial_space_utils/svd_method_of_snapshots.py]: can you please do this in a spearate issue

Comment (cwschilly) [reply romtools/trial_space_utils/svd_method_of_snapshots.py]: https://github.com/Pressio/rom-tools-and-workflows/pull/60

**Pull request merged**